language/move-command-line-common/src/character_sets.rs

```diff
@@ -1,32 +1,70 @@
 // Copyright (c) The Diem Core Contributors
 // SPDX-License-Identifier: Apache-2.0
 
-/// Determine if a character is an allowed eye-visible (printable) character.
+/// Determine if a character is horizontal whitespace.
 ///
-/// The only allowed printable characters are the printable ascii characters (SPACE through ~) and
-/// tabs. All other characters are invalid and we return false.
-pub fn is_permitted_printable_char(c: char) -> bool {
-    let x = c as u32;
-    let is_above_space = x >= 0x20; // Don't allow meta characters
-    let is_below_tilde = x <= 0x7E; // Don't allow DEL meta character
-    let is_tab = x == 0x09; // Allow tabs
-    (is_above_space && is_below_tilde) || is_tab
+/// Unicode standard 6.2, chapter 6.2, table 6-2 provides an inexhaustive list
+/// of 18 kinds of horizontal space characters, but there are many more. This
+/// function uses the table of Unicode whitespace on Wikipedia as a reference:
+/// https://en.wikipedia.org/wiki/Whitespace_character#Unicode
```

Comment:
Maybe we should point to https://www.unicode.org/Public/UCD/latest/ucd/PropList.txt ?